macros/src/main/scala-2/com/softwaremill/macwire/internals/CompanionCrimper.scala

```diff
@@ -32,12 +32,12 @@ private[macwire] class CompanionCrimper [C <: blackbox.Context, T: C#WeakTypeTag
 
   lazy val applyParamLists: Option[List[List[Symbol]]] = apply.map(_.asMethod.paramLists)
 
-  def wireParams(paramLists: List[List[Symbol]]): List[List[Tree]] = paramLists.map(_.map(p => dependencyResolver.resolve(p, p.typeSignature)))
+  def wireParams(dependencyResolver: DependencyResolverType)(paramLists: List[List[Symbol]]): List[List[Tree]] = paramLists.map(_.map(p => dependencyResolver.resolve(p, p.typeSignature)))
```

Comment:
can't we have `dependencyResolver` as a parameter of `CompanionCrimper`?

Reply:
I tried to define CompanioCrimper like this:
```Scala
private[macwire] class CompanionCrimper [C <: blackbox.Context, T: C#WeakTypeTag](val c: C, log: Logger)(dependencyResolver: DependencyResolver[c.type, c.universe.Type, c.universe.Tree]) {
...
}
```

but it fails with 
```
[error][...]/macwire/macros/src/main/scala-2/com/softwaremill/macwire/internals/CompanionCrimper.scala:5:145: not found: value c
[error] private[macwire] class CompanionCrimper [C <: blackbox.Context, T: C#WeakTypeTag](val c: C, log: Logger)(dependencyResolver: DependencyResolver[c.type, c.universe.Type, c.universe.Tree]) {
[error]                                                                                                                                                 ^
[error][...]/macwire/macros/src/main/scala-2/com/softwaremill/macwire/internals/CompanionCrimper.scala:5:153: not found: value c
[error] private[macwire] class CompanionCrimper [C <: blackbox.Context, T: C#WeakTypeTag](val c: C, log: Logger)(dependencyResolver: DependencyResolver[c.type, c.universe.Type, c.universe.Tree]) {
[error]                                                                                                                                                         ^
[error][...]/macwire/macros/src/main/scala-2/com/softwaremill/macwire/internals/CompanionCrimper.scala:5:170: not found: value c
[error] private[macwire] class CompanionCrimper [C <: blackbox.Context, T: C#WeakTypeTag](val c: C, log: Logger)(dependencyResolver: DependencyResolver[c.type, c.universe.Type, c.universe.Tree]) {
```

Reply:
ah right, you can't use dependent types in constructors, forgot about that :)